services/user/Evaluations/plugin/src/consensus.rs

```diff
@@ -0,0 +1,109 @@
+use average::Variance;
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+pub fn remove_eliminated_options(submissions: Vec<Vec<u8>>, options_size: u8) -> Vec<Vec<u8>> {
+    let elimination_threshold = ((options_size as u16 * 2) / 3) as u8;
+
+    let unranked_counts: HashMap<u8, u8> = submissions
+        .iter()
+        .flat_map(|ranked| {
+            (1..options_size + 1)
+                .filter(|o| !ranked.contains(o))
+                .collect::<Vec<u8>>()
+        })
+        .fold(HashMap::new(), |mut acc, option| {
+            *acc.entry(option).or_insert(0) += 1;
+            acc
+        });
+    
+    let eliminated_options: Vec<u8> = unranked_counts
+        .into_iter()
+        .filter(|(_, count)| count >= &elimination_threshold)
+        .map(|(option, _)| option)
+        .collect();
+
+    submissions
+        .into_iter()
+        .map(|ranked| {
+            ranked
+                .into_iter()
+                .filter(|o| !eliminated_options.contains(o))
+                .collect()
+        })
+        .collect()
+}
+
+pub fn get_level(option: &u8, ranked_options: &Vec<u8>, options_size: u8) -> u8 {
+    ranked_options
+        .iter()
+        .position(|o| o == option)
+        .map(|i| ((options_size as usize) - i) as u8)
+        .unwrap_or(0)
+}
+
+pub fn calculate_consensus(
+    submissions: Vec<Option<Vec<u8>>>,
```

Comment:
Would be more pure if you filter submissions outside this function, so it just takes `Vec<Vec<u8>>`